jdaviz/cli.py

```diff
@@ -76,12 +75,37 @@ def main(filepaths=None, layout='default', instrument=None, browser='default',
         args += ['--auto-restart']
     else:
         args += ['--production']
-    cli(['run', 'jdaviz.solara',
-         '--theme-loader', 'plain',
-         '--theme-variant', theme,
-         '--host', host,
-         '--port', port] + args)
-
+    run_solara(host=host, port=port, theme=theme, browser=browser)
+
+
+def run_solara(host, port, theme, browser):
+    os.environ["SOLARA_APP"] = "jdaviz.solara"
+    import solara.server.starlette
+    import solara.server.settings
+    solara.server.settings.theme.variant = theme
+    solara.server.settings.theme.loader = "plain"
+
+    server = solara.server.starlette.ServerStarlette(host="localhost", port=port)
+    print(f"Starting server on {server.base_url}")
+    server.serve_threaded()
+    server.wait_until_serving()
+    if browser == "qt":
+        from qtpy.QtWidgets import QApplication
+        from qtpy.QtWebEngineWidgets import QWebEngineView
+        from qtpy import QtCore
+
+        app = QApplication([""])
+        web = QWebEngineView()
+        web.resize(1000, 1000)
+        web.setUrl(QtCore.QUrl(server.base_url))
+        web.show()
+
+        app.exec_()
+    else:
+        import webbrowser
+        controller = webbrowser.get(None if browser == 'default' else browser)
+        controller.open(server.base_url)
+        server.join()
```

Comment:
Add a blank line to make flake8 happy.

```suggestion
        server.join()

```